mypyc/test-data/irbuild-any.test

```diff
@@ -198,3 +198,18 @@ L0:
     b = r4
     return 1
 
+[case testFunctionBasedOps]
```

Comment:
Once this is merged, I'll [submit a PR for `__pow__` and co](https://github.com/python/mypy/commit/c408b06ee66aa0be0b991e8b1096b680c9476a2d). which depends on this patch.